File: frontend/src/types/project.ts
export interface projectModel {
    projectID: string
    slug: string
    displayName: string
    description: string
    createdAt: string
    updatedAt: string
}

export interface projectRefModel {
    projectID: string
    slug: string
    displayName: string
}

export interface createProjectModel {
    slug: string
    displayName: string
    description?: string
}


export interface projectMemberModel {
    projectID: string
    userID: string
    username: string
    email: string
    fullname: string
    phone: string
    projectRole: string
    createdAt: Date
}

File: internal/db/entities/project.go
package entities

import "time"

type Project struct {
	Base
	Slug                 string `gorm:"type:varchar(64);uniqueIndex;not null"`
	Name                 string `gorm:"type:varchar(128);not null"`
	Description          string `gorm:"type:text"`
	CollaborationEnabled bool   `gorm:"type:boolean;not null;default:false"`
}

type ProjectMember struct {
	ID          string    `gorm:"type:varchar(36);primaryKey"`
	CreatedAt   time.Time `gorm:"autoCreateTime"`
	UpdatedAt   time.Time `gorm:"autoUpdateTime"`
	ProjectID   string    `gorm:"type:varchar(36);index:idx_project_members_project_id;uniqueIndex:idx_project_members_project_user;not null"`
	UserID      string    `gorm:"type:varchar(36);index:idx_project_members_user_id;uniqueIndex:idx_project_members_project_user;not null"`
	ProjectRole string    `gorm:"type:varchar(16);not null"`
}
